src/c1_foeppl_von_karman/foeppl_von_karman_curvable_bell_elements.h

```diff
@@ -46,672 +46,1160 @@
+    }
+
+    /// C-style output function:
+    ///   x,y,u   or    x,y,z,u at n_plot*(n_plot+1)/2 plot points
+    void output(FILE* file_pt, const unsigned& n_plot)
+    {
+      FoepplVonKarmanEquations::output(file_pt, n_plot);
+    }
+
+
+    /// Output function for an exact solution:
+    ///  x,y,u_exact   or    x,y,z,u_exact at n_plot*(n_plot+1)/2 plot points
+    void output_fct(std::ostream& outfile,
+                    const unsigned& n_plot,
+                    FiniteElement::SteadyExactSolutionFctPt exact_soln_pt)
+    {
+      FoepplVonKarmanEquations::output_fct(outfile, n_plot, exact_soln_pt);
+    }
+
+    /// Output function for a time-dependent exact solution.
+    ///  x,y,u_exact   or    x,y,z,u_exact at n_plot*(n_plot+1)/2 points
+    /// (Calls the steady version)
+    void output_fct(std::ostream& outfile,
+                    const unsigned& n_plot,
+                    const double& time,
+                    FiniteElement::UnsteadyExactSolutionFctPt exact_soln_pt)
+    {
+      FoepplVonKarmanEquations::output_fct(
+        outfile, n_plot, time, exact_soln_pt);
+    }
+
+
+    //----------------------------------------------------------------------
+    // Jacobian and residual contributions
+
+    /// Add the element's contribution to its residual vector (wrapper) with
+    /// cached association matrix
+    void fill_in_contribution_to_residuals(Vector<double>& residuals)
+    {
+      // Store the expensive-to-construct matrix
+      this->store_association_matrix();
+      // Call the generic routine with the flag set to 1
+      FoepplVonKarmanEquations::fill_in_contribution_to_residuals(residuals);
+      // Remove the expensive-to-construct matrix
+      this->delete_association_matrix();
+    }
+
+    /// Add the element's contribution to its residual vector and
+    /// element Jacobian matrix (wrapper) with caching of association matrix
+    void fill_in_contribution_to_jacobian(Vector<double>& residuals,
+                                          DenseMatrix<double>& jacobian)
+    {
+      // Store the expensive-to-construct matrix
+      this->store_association_matrix();
+      // Call the generic routine with the flag set to 1
+      FoepplVonKarmanEquations::fill_in_contribution_to_jacobian(residuals,
+                                                                 jacobian);
+      // Remove the expensive-to-construct matrix
+      this->delete_association_matrix();
+    }
+
+
+    //----------------------------------------------------------------------
+    // Geometry and boundaries
+
+   
+   /// Clamp: i.e. pin the in-plane displacements and pin the out-of-plane
+   /// displacement and its normal derivative. We also apply implied
+   /// boundary conditions (e.g. specification of dw/dn also implies
+   /// d^2w/dn/dzeta etc. boundary_values_pt[i] describes boundary conditions
+   /// for the three displacement components... hierher Aidan: this is where
```

Comment:
It might already be explained elsewhere.

Displacements are unrotated, just the coordinates:
    (u_x,u_y,u_z)(n,t)

Reply:
clarified.